### ISSUE TITLE
[91ed74a] safe_parse_pubkey

### ISSUE THREAD

**Issue by lzsaver** [spesmilo/electrum#2969]
```
Traceback (most recent call last):
  File "/home/user/electrum/lib/python3.5/site-packages/electrum_gui/qt/main_window.py", line 1337, in do_send
    tx = self.wallet.make_unsigned_transaction(coins, outputs, self.config, fee)
  File "/home/user/electrum/lib/python3.5/site-packages/electrum/wallet.py", line 858, in make_unsigned_transaction
    fee = fee_estimator(tx.estimated_size())
  File "/home/user/electrum/lib/python3.5/site-packages/electrum/util.py", line 213, in <lambda>
    return lambda *args, **kw_args: do_profile(func, args, kw_args)
  File "/home/user/electrum/lib/python3.5/site-packages/electrum/util.py", line 209, in do_profile
    o = func(*args, **kw_args)
  File "/home/user/electrum/lib/python3.5/site-packages/electrum/transaction.py", line 826, in estimated_size
    weight = self.estimated_weight()
  File "/home/user/electrum/lib/python3.5/site-packages/electrum/transaction.py", line 854, in estimated_weight
    total_tx_size = self.estimated_total_size()
  File "/home/user/electrum/lib/python3.5/site-packages/electrum/transaction.py", line 837, in estimated_total_size
    return len(self.serialize(True)) // 2 if not self.is_complete() or self.raw is None else len(self.raw) // 2  # ASCII hex string
  File "/home/user/electrum/lib/python3.5/site-packages/electrum/transaction.py", line 773, in serialize
    txins = var_int(len(inputs)) + ''.join(self.serialize_input(txin, self.input_script(txin, estimate_size)) for txin in inputs)
  File "/home/user/electrum/lib/python3.5/site-packages/electrum/transaction.py", line 773, in <genexpr>
    txins = var_int(len(inputs)) + ''.join(self.serialize_input(txin, self.input_script(txin, estimate_size)) for txin in inputs)
  File "/home/user/electrum/lib/python3.5/site-packages/electrum/transaction.py", line 672, in input_script
    scriptSig = bitcoin.p2wpkh_nested_script(pubkeys[0])
  File "/home/user/electrum/lib/python3.5/site-packages/electrum/bitcoin.py", line 336, in p2wpkh_nested_script
    pubkey = safe_parse_pubkey(pubkey)
NameError: name 'safe_parse_pubkey' is not defined
```

**Comment by SomberNight** [comment]
See #2961

**Comment by lzsaver** [comment]
Thanks. I did not notice your last comment.

**Comment by SomberNight** [comment]
Temporary patch:
```
diff --git a/lib/bitcoin.py b/lib/bitcoin.py
index 1d8c5acc..f85c3e39 100644
--- a/lib/bitcoin.py
+++ b/lib/bitcoin.py
@@ -332,6 +332,13 @@ def public_key_to_p2wpkh(public_key):
 def script_to_p2wsh(script):
     return hash_to_segwit_addr(sha256(bfh(script)))
 
+def safe_parse_pubkey(x):
+    from . import keystore
+    try:
+        return keystore.xpubkey_to_pubkey(x)
+    except:
+        return x
+
 def p2wpkh_nested_script(pubkey):
     pubkey = safe_parse_pubkey(pubkey)
     pkh = bh2u(hash_160(bfh(pubkey)))
diff --git a/lib/transaction.py b/lib/transaction.py
index 658ff593..2e89ca6f 100644
--- a/lib/transaction.py
+++ b/lib/transaction.py
@@ -287,12 +287,6 @@ def match_decoded(decoded, to_match):
 def parse_sig(x_sig):
     return [None if x == NO_SIGNATURE else x for x in x_sig]
 
-def safe_parse_pubkey(x):
-    try:
-        return xpubkey_to_pubkey(x)
-    except:
-        return x
-
 def parse_scriptSig(d, _bytes):
     try:
         decoded = [ x for x in script_GetOp(_bytes) ]

```

**Comment by lzsaver** [comment]
Your workaround works for me. Thank you.

**Comment by ecdsa** [comment]
fixed in 9f6e174b4ffd5bff6280b205b9168ac765c27c80
